src/wp-admin/includes/file.php

```diff
@@ -1946,6 +1946,76 @@ function copy_dir( $from, $to, $skip_list = array() ) {
 	return true;
 }
 
+/**
+ * Moves a directory from one location to another.
+ *
+ * Recursively invalidates OPcache on success.
+ *
+ * If the renaming failed, falls back to copy_dir().
+ *
+ * Assumes that WP_Filesystem() has already been called and setup.
+ *
+ * @since 6.2.0
+ *
+ * @global WP_Filesystem_Base $wp_filesystem WordPress filesystem subclass.
+ *
+ * @param string $from      Source directory.
+ * @param string $to        Destination directory.
+ * @param bool   $overwrite Optional. Whether to overwrite the destination directory if it exists.
+ *                          Default false.
+
+ * @return true|WP_Error True on success, WP_Error on failure.
+ */
+function move_dir( $from, $to, $overwrite = false ) {
+	global $wp_filesystem;
+
+	if ( ! $overwrite && $wp_filesystem->exists( $to ) ) {
+		return new WP_Error(
+			'destination_already_exists_move_dir',
+			sprintf(
+				/* translators: %s: The destination directory path. */
+				__( 'The destination folder, %s, already exists.' ),
+				"<code>$to</code>"
+			)
+		);
+	}
+
```

Comment:
I recently encountered an edge case where `$from` and `$to` differed only by a trailing slash. This resulted in the clearing of `$to` prior to `rename()` deleting the source. We could easily add a guard for that case as in the following.

```php
	if ( trailingslashit( $from ) === trailingslashit( $to ) ) {
		return false;
	}
```

Would need to update `@return` also.